### PR TITLE
feat(job): support built-in runtime in model package

### PR DESCRIPTION
**Pull request by goldenxinxing**
## Description

1. [client]copy runtime package to server in client when copy a model package with built-in runtime
2. [client]fix the exception(`Only one live display may be active at once)` when use Progress multiples in the same process
3. [server]add a column named 'built_in_runtime' for model version
4. [console]add built-in runtime
5. [e2e]add e2e test case
image
image

## Modules
- [x] UI
- [x] Controller
- [ ] Agent
- [x] Client
- [ ] Python-SDK
- [ ] Others

## Checklist
- [x] run code format and lint check
- [x] add unit test
- [ ] add necessary doc

### CODE REVIEW

File: console/src/domain/runtime/components/RuntimeTreeSelector.tsx [hunk 1]
```diff
@@ -19,12 +19,12 @@ const RuntimeTreeNode = themedStyled('div', () => ({
...
     const [t] = useTranslation()
-    const { projectId } = props
+    const { projectId, builtIn } = props
     const runtimeInfo = useFetchRuntimeTree(projectId)
     const $treeData = React.useMemo(() => {
-        if (!runtimeInfo.isSuccess) return []
-
-        const treeData: TreeNodeData[] = runtimeInfo.data.map((runtime) => {
+        if (!runtimeInfo.isSuccess) return builtIn? [builtIn] : []
+        let mergedItems = builtIn ? [builtIn, ...runtimeInfo.data.filter(i => i.versions[0].id !== builtIn.versions[0].id)] : runtimeInfo.data
```
Comment (jialeicui) [inline console/src/domain/runtime/components/RuntimeTreeSelector.tsx:26]: The corresponding version may not be listed first?

File: server/controller/src/main/java/ai/starwhale/mlops/domain/model/ModelService.java [hunk 2]
```diff
@@ -622,9 +659,62 @@ public void uploadSrc(Long modelVersionId, MultipartFile multipartFile, ModelUpl
...
+                        new CustomMultipartFile(fileName, storageAccessService.get(filePath)), runtimeRequest);
+                    return true;
+                } else {
+                    log.warn("built-in runtime upload error, no path found");
+                }
+            }
+        } catch (Exception e) {
+            log.warn("sync built-in runtime error for model version:{},error:{}", modelVersionEntity.getId(), e);
+        }
+        return false;
```
Comment (jialeicui) [inline server/controller/src/main/java/ai/starwhale/mlops/domain/model/ModelService.java:715]: Would it be better for the runtime to let the client upload a copy and tell the relationship?

File: console/src/domain/runtime/components/RuntimeTreeSelector.tsx (hunk 1 above)
Comment (goldenxinxing) [reply console/src/domain/runtime/components/RuntimeTreeSelector.tsx]: yes, fixed

File: server/controller/src/main/java/ai/starwhale/mlops/domain/model/ModelService.java (hunk 2 above)
Comment (goldenxinxing) [reply server/controller/src/main/java/ai/starwhale/mlops/domain/model/ModelService.java]: done

File: client/starwhale/base/bundle_copy.py [hunk 3]
```diff
@@ -54,6 +54,18 @@ class _UploadPhase:
 
 
 class BundleCopy(CloudRequestMixed):
+    progress: Progress = Progress(
```
Comment (tianweidut) [inline client/starwhale/base/bundle_copy.py:57]: we need Progress context manager : https://rich.readthedocs.io/en/stable/progress.html#advanced-usage

```
With Progress(xxx) as p:
   ...
```

Comment (goldenxinxing) [reply client/starwhale/base/bundle_copy.py]: fixed

**Pull request merged**